### DISCUSSION TITLE
Reuse assignment

### DISCUSSION THREAD

**Discussion by pinventado** [community/community#45016] [GitHub Education]
How do we use the "reuse assignment" feature? I am able to change the organization, but the classroom dropdown does not show any results.

I've tried creating a classroom under the same organization, but it still doesn't work. Any ideas?

<img width="488" alt="image" src="https://user-images.githubusercontent.com/341001/213885195-e71d3337-9075-4554-8737-0b1d17e564f7.png">

**Comment by jge162** [comment]
Hey @pinventado first and foremost. Go Titans!! 

The "reuse assignment" feature in GitHub Classroom allows you to create a new assignment by reusing the settings of an existing assignment. In order to use this feature, you need to have the appropriate permissions in the organization where you want to create the new assignment.

To use the "reuse assignment" feature, you will first need to navigate to the organization where you want to create the new assignment. Then, you will need to select the classroom that contains the assignment you want to reuse. Once you have selected the classroom, you should see a list of all the assignments that are currently in that classroom. You can then click the "Reuse" button next to the assignment you want to reuse.

If you are not able to see the classroom in the dropdown, check if the classroom is private and you have permission to access the classroom. If you've created a classroom in the same organization but it's not appearing, please check if you have selected the correct organization.

Also ensure you have the correct permissions to reuse the assignments in the classroom, if you're an owner or teacher of the classroom you should be able to reuse the assignments. If you still face issues, please reach out to GitHub support for assistance.

**Comment by pinventado** [comment]
Thanks, @jge162 ! Go titans :)

Hmm, here are some things I checked, which I thought should give me full access

Organization - I am currently set as owner
2022 Fall classroom - I am listed as a classroom admin and it is associated with the organization
2023 Spring classroom - I am listed as a classroom admin and it is associated with the same organization

I wasn't sure what you meant about checking if the classroom is private. Is there a setting for that and where could I find it?

Thanks for your help!

**Reply by jge162** [reply]
Like is the repository private or public is what I meant. 

Regarding the "private" classroom setting, it could be that the classroom you are trying to reuse the assignment in has been set to "private" or "hidden" by the administrator. In some LMSs, private classrooms are not visible to users without permission to access them.

You could check with the administrator or the support team of your LMS to confirm if the classroom is set to private or hidden, and if so, if you have permission to access it.

btw LMS stands for Learning Management System.

**Comment by pinventado** [comment]
I see. So on lab 1 of the 2022 Fall classroom, I see that the repository visibility was set to private.

<img width="640" alt="image" src="https://user-images.githubusercontent.com/341001/213888149-ada4e96d-c501-4eeb-9cfe-e5aa63637af9.png">

I tried switching it to public then tried "reuse assignment" again, but still couldn't find the 2023 Spring classroom in the dropdown. I don't have anything inside the 2023 Spring classroom yet.

**Reply by jge162** [reply]
So u got it now? Sounds like you do.

**Reply by jge162** [reply]
Are u taking 131 this semester?

**Reply by pinventado** [reply]
No, it didn't work unfortunately. I still can't find the 2023 Spring classroom in the dropdown.

**Reply by pinventado** [reply]
No, I'm not. I'm teaching 121 this semester though :)

**Comment by jge162** [comment]
Oh ok, perfect. 

Hm still not visible. The classroom may not have been created yet in the LMS. Double-check with the administrator or the support team to confirm that the classroom has been set up and is active.

also maybe a filter applied to the dropdown menu is hiding the 2023 Spring classroom. Try removing any filters that may be applied to the menu to see if the classroom becomes visible.

And maybe the classroom may be associated with a different organization within the LMS. Double-check that the classroom is associated with the same organization as the assignment you are trying to reuse.

**Reply by pinventado** [reply]
I don't think IT enabled github classroom so its not integrated as far as I know. Both classrooms were not connected to the LMS. 

What do you mean filters? I see a textbox for the dropdown and made sure nothing was typed in. Is that what you mean?

<img width="425" alt="image" src="https://user-images.githubusercontent.com/341001/213888494-b8664ee0-da14-4cc8-82a6-4d7a31b0c918.png">

I double-checked and they are in the same organization. Unless there is something in the IT side of things that I can't see or access.

**Reply by jge162** [reply]
A dropdown menu with options such as "All," "Active," or "Inactive", It's also possible that the LMS you are using doesn't have any built-in filters or search functionality, in that case, I apologize for any confusion. I usually am on the student side so idk exactly. Just mentioned to be safe. 

I’m thinking it’s an IT issue now. Might have to reach out.

**Reply by pinventado** [reply]
Oh wait. I may have missed to mention that I am doing all this on GitHub classroom directly. I'm not doing anything inside the LMS.

Were you using an LMS in your case?

**Reply by jge162** [reply]
Yes

**Reply by pinventado** [reply]
I see. I should have started with that. There may be settings I'm not aware of causing some issues ... or at least unless there's something else. Will try to reach out to Github support.

Thanks for all the help and good luck with the new semester!

**Reply by jge162** [reply]
Tysm. It’s my final semester finally. Taking my last class. Have a great semester as well!!!
